limesurvey/limesurvey.py

```diff
@@ -61,8 +61,14 @@ def student_view(self, context=None):
         if context:
             pass  # TO-DO: do something based on the context.
 
-        anonymous_user_id = self.runtime.anonymous_student_id
-        self.add_participant_to_survey(self.runtime.get_real_user(anonymous_user_id), anonymous_user_id)
+        limesurvey_api_url = getattr(settings, "LIMESURVEY_INTERNAL_API", None)
+
+        if limesurvey_api_url:
+            anonymous_user_id = self.runtime.anonymous_student_id
+            self.add_participant_to_survey(
+                self.runtime.get_real_user(anonymous_user_id),
+                anonymous_user_id,
+            )
```

Comment:
why doing this here? do we use `limesurvey_api_url` anywhere here? I think it's best if we check whether the API URL it's specified just in `self_invoke`. The other method calls should be agnostic of the setting existing

Reply:
Without this validation the view of studio fails. Try access to `user.profile.name`, but `user` is `None`
![image](https://github.com/eduNEXT/xblock-limesurvey/assets/64033729/16eb2d31-cb65-4f33-ab69-d99918542035)

Reply:
But it would be checked inside the _invoke. Isn't that the same? Try refactoring so that's the first thing it's checked

Reply:
I have performed the refactoring and the problem persists. However, I have added a new validation to know if the block is being rendered from the Studio view. If so, the block only renders a simple message, and avoids consuming other methods and the LimeSurvey API.